### ISSUE TITLE
Usage with base path set in a config

### ISSUE THREAD

**Issue by dallyh** [jlarmstrongiv/astro-i18n-aut#21]
I cannot get my head around on how to use this integration with base path. Everything works fine when the base path is not set, that is when the page is located for ex. under http://localhost:4321/projects/, but with base path, no locales get resolved and return 404 - for ex. http://localhost:4321/website-concept/en/projects/.

![image](https://github.com/jlarmstrongiv/astro-i18n-aut/assets/6968534/2efdeaf5-7dfc-40e4-991d-f791d7e75559)

My config for locales is set like this:
```typescript

import { defineConfig } from "astro/config";
import { i18n } from "astro-i18n-aut/integration";
import react from "@astrojs/react";

const defaultLocale = "cs";
const locales = {
    en: "en-US", 
    cs: "cs-CZ",
};

// https://astro.build/config
export default defineConfig({
    site: "https://some-website.com",
    base: "/website-concept",
    trailingSlash: "always",
    integrations: [
        react(),
        i18n({
            locales,
            defaultLocale,
        }),
    ],
});
```

Thanks for any help with this!

**Comment by jlarmstrongiv** [comment]
@dallyh do you have a github repo I could look at?  

I’d like to see whether this is part of a known Astro V3 [bug](https://github.com/jlarmstrongiv/astro-i18n-aut/issues/19) or something else.  

I wrote support for [base](https://github.com/jlarmstrongiv/astro-i18n-aut/blob/main/src/integration/integration.ts#L119) paths, but there may be a bug in the implementation.

**Comment by dallyh** [comment]
@jlarmstrongiv I've just created one here: https://github.com/dallyh/astro-i18n-base-path

The issue seems to persists even on build, as when I build it with base path set and then run preview, the localized pages always return 404.

**Comment by seanpue** [comment]
In the demo uploaded, the base path is being duplicated, so this works: http://localhost:4321/example-base/example-base/es/. It appears ``config.base`` is not necessary [here](https://github.com/jlarmstrongiv/astro-i18n-aut/blob/main/src/integration/integration.ts#L119).

**Comment by jlarmstrongiv** [comment]
Thank you @dallyh and @seanpue!  

@seanpue you are right, Astro automatically adds `config.base`

@dallyh thank you for the reproduction!  I have fixed the paths in `0.3.5` and have added `baseUrl` support to all helper functions too

Please let me know if either of you run into any more issues with `base`!

**Comment by jlarmstrongiv** [comment]
@seanpue and @dallyh 

I realized that some functions like middleware didn’t support `BASE_URL`.  the latest version `0.4.14` should have fixed it for all functions.  

This major upgrade also renamed some helper functions and removed all params from middleware.  I can now import the `defaultLocale` and `BASE_URL` into helper functions and middleware with the new virtual modules implementation.
